pkg/kubelet/kubelet_node_status.go

```diff
@@ -394,6 +405,11 @@ func (kl *Kubelet) tryUpdateNodeStatus(tryNumber int) error {
 	if err != nil {
 		return err
 	}
+
+	if kl.getNodeReadyLastHeartbeatTime(updatedNode) != nil {
```

Comment:
@wojtek-t this bakes in assumptions about what `kl.setNodeStatus` does (that it updates the heartbeat timestamp of the node object). we know those assumptions are desired to change in the future (xref https://github.com/kubernetes/community/pull/2090). is @kubernetes/sig-node-pr-reviews ok with entangling this function somewhat in the short-term to get a performance gain at the cost of having to untangle it when node heartbeats get moved to a separate function?

Reply:
I'm not 100% sure that we will have to change that when implementing that proposal. But my personal feeling is that even if we have too, this change is small enough that it's worth it. WDYT?

Reply:
/cc @mtaufen, who's working on the heartbeat proposal.